tools/vitest-config-factory.ts

```diff
@@ -0,0 +1,151 @@
+/// <reference types="vitest" />
+import { pathToFileURL } from 'node:url';
+import { defineConfig, mergeConfig } from 'vite';
+import type { UserConfig as ViteUserConfig } from 'vite';
+import type { CoverageOptions } from 'vitest';
+import { setupPresets } from './vitest-setup-presets.js';
+import { tsconfigPathAliases } from './vitest-tsconfig-path-aliases.js';
+
+export type TestKind = 'unit' | 'int' | 'e2e';
+
+export interface CreateVitestConfigOptions {
+  projectKey: string;
+  kind: TestKind;
+  projectRoot: string | URL;
+  include?: string[];
+  setupFiles?: string[];
+  /** If true, the factory will not inject the baseline setupFiles for the given kind. */
+  overrideSetupFiles?: boolean;
+  globalSetup?: string[];
+  coverage?: {
+    enabled?: boolean;
+    exclude?: string[];
+    reportsSubdir?: string;
+  };
+  testTimeout?: number;
+  typecheckInclude?: string[];
+  cacheKey?: string;
+}
+
+export function createVitestConfig(
+  options: CreateVitestConfigOptions,
+): ViteUserConfig {
```

Comment:
As mentioned in [this comment](https://github.com/code-pushup/cli/pull/1107/files#r2324898492) I would use a API and more important data structure/types as lose as possible to vitest.